crates/engine/tree/src/tree/mod.rs

```diff
@@ -2281,12 +2269,11 @@ where
 
         let persistence_not_in_progress = !self.persistence_state.in_progress();
 
-        let state_root_result = std::thread::scope(|scope| {
```

Comment:
do we need this to be scoped?

Reply:
i don't think so, this was introduced to be able to create `ProofBlindedProviderFactory`